File: src/main/java/org/flossware/jnexus/ComponentMetadata.java
package org.flossware.jnexus;

import java.time.Instant;

/**
 * Represents detailed metadata for a repository component from Nexus.
 * <p>
 * This record extends beyond basic component information to include
 * metadata such as creation dates, content types, and checksums.
 * This enriched data enables advanced filtering, auditing, and analysis.
 * </p>
 *
 * @param id           the unique identifier of the component in Nexus
 * @param path         the full path to the component artifact
 * @param fileSize     the size of the component file in bytes
 * @param contentType  the MIME type of the component (e.g., "application/java-archive")
 * @param format       the repository format (e.g., "maven2", "npm", "docker")
 * @param createdDate  the timestamp when the component was uploaded to Nexus (null if not available)
 * @param lastModified the timestamp when the component was last modified (null if not available)
 * @param checksum     the primary checksum of the component (SHA1 or MD5, null if not available)
 * @author sfloess
 * @since 1.0
 */
public record ComponentMetadata(
    String id,
    String path,
    long fileSize,
    String contentType,
    String format,
    Instant createdDate,
    Instant lastModified,
    String checksum
) {
    /**
     * Converts this metadata record to a simple RepoRecord.
     * <p>
     * Useful for backward compatibility when only basic information is needed.
     * </p>
     *
     * @return a RepoRecord containing id, fileSize, and path
     */
    public RepoRecord toRepoRecord() {
        return new RepoRecord(id, fileSize, path);
    }
}

File: jnexus-ios/Shared/Models/ComponentMetadata.swift
//
//  ComponentMetadata.swift
//  JNexus
//
//  Enhanced component record with full metadata
//  Mirrors the Java ComponentMetadata from jnexus-core
//

import Foundation

/// Component record with detailed metadata
struct ComponentMetadata: Codable, Identifiable, Hashable {
    /// Unique component identifier
    let id: String

    /// File size in bytes
    let fileSize: Int64

    /// Component path in repository
    let path: String

    /// MIME content type (e.g., "application/java-archive")
    let contentType: String?

    /// Repository format (e.g., "maven2", "npm", "docker")
    let format: String?

    /// When component was created/uploaded
    let createdDate: Date?

    /// When component was last modified
    let lastModified: Date?

    /// Checksum (SHA1 preferred, MD5 fallback)
    let checksum: String?

    /// CodingKeys for JSON serialization
    enum CodingKeys: String, CodingKey {
        case id
        case fileSize
        case path
        case contentType
        case format
        case createdDate
        case lastModified
        case checksum
    }
}
